### PR TITLE
:bug: Source MongoDB - Added milliseconds to convert timestamp to datetime format

### LINKED ISSUE #7556: Source: MongoDB - Timestamp microseconds missing in Cursor - Incremental change capture
## Enviroment
- **Airbyte version**:  v0.30.24-alpha
- **OS Version / Instance**: Ubuntu 20.04 lts
- **Deployment**: Docker (in local)
- **Source Connector and version**: airbyte/source-mongodb-v2:0.1.3
- **Destination Connector and version**: S3 - JSONL
- **Severity**: Critical
- **Step where error happened**: Sync job

## Current Behavior
Incremental change capture (cursor) is missing microseconds. Due to this same records are being captured repeatedly. 
For example, Lets assume, I have chosen "UpdatedAt" field as cursor. The last processed record "UpdatedAt" timestamp is "2021-11-01 10:59:01.567Z". But the cursor is being captured as "2021-11-01 10:59:01Z". Eventhough there is no changes done at the source, in the next run, the same set of record is captured again because its checking for the records GREATER THAN "2021-11-01 10:59:01Z" so obviously the record which has timestamp "2021-11-01 10:59:01.567Z" is processed again.

## Expected Behavior
The cursor should capture the exact data with microseconds (in the above example case, it should capture the cursor as "2021-11-01 10:59:01.567Z")

## Steps to Reproduce
1. Create a collection and Insert some records in MongoDB. Make sure record has a timestamp field that will be used as a cursor and also the timestamp has microseconds.
2. Configure source (MongoDb), destination(S3), and connector as Incremental Append. selected a timestamp field as cursor.
3. Run the connector or run docker run --rm --init -i -v airbyte_workspace:/data -v /tmp/airbyte_local:/local -w /data/{workspace} --network host --log-driver none airbyte/source-mongodb-v2:0.1.0 read --config source_config.json --catalog source_catalog.json
4. Rerun the connection again without making any changes in source. This should process the record again.

## Are you willing to submit a PR? 
Since I'm not proficient in Java, I'm unable to figure the issue in code.

### PR DESCRIPTION
**Pull request by andriikorotkov**
## What
Added milliseconds to convert timestamp to datetime format.

- Root case - the pattern we are using for transformation did not support milliseconds.
- I was able to reproduce the bug locally.
- I added a new pattern and a new method for converting timestamp.

## Pre-merge Checklist
Expand the relevant checklist and delete the others. 

<details><summary> <strong> New Connector </strong></summary>
<p>

#### Community member or Airbyter
   
- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/SUMMARY.md`
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
</p>
</details>

<details><summary> <strong> Updating a connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the new connector version is published, connector version bumped in the seed directory as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</p>
</details>

<details><summary> <strong> Connector Generator </strong> </summary>
<p>
   
- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed.
</p>
</details>

### CODE REVIEW

**Review by alexandr-shegeda — APPROVED** [review]
LGTM. we can proceed with publish/merge without airbyte review

File: airbyte-db/lib/src/main/java/io/airbyte/db/DataTypeUtils.java [hunk 1]
```diff
@@ -37,6 +40,10 @@
     }
   }
 
+  public static String toISO8601StringWithMilliseconds(final long epochMillis) {
+    return DATE_FORMAT_WITH_MILLISECONDS.format(Date.from(Instant.ofEpochMilli(epochMillis)));
```
Comment (VitaliiMaltsev) [inline airbyte-db/lib/src/main/java/io/airbyte/db/DataTypeUtils.java:44]: from task definition: _The cursor should capture the exact data with microseconds_
Not milliseconds

**Pull request merged**